### PR TITLE
Haar 3284: prison api client retries

### PR DESCRIPTION
**Pull request by DaveLlewellynMoJ**
Added client retry policy to prison API client requests + tests.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/client/PrisonApiClient.kt [hunk 1]
```diff
@@ -1,37 +1,61 @@
...
+    } catch (ex: ClientAuthorizationException) {
+      throw FatalSubjectAccessRequestException(
+        message = "prisonApiClient error authorization exception",
+        cause = ex,
+        event = ACQUIRE_AUTH_TOKEN,
+        subjectAccessRequestId = subjectAccessRequest.id,
+        params = mapOf(
+          "cause" to ex.cause?.message,
+        ),
+      )
```
Comment (DaveLlewellynMoJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/client/PrisonApiClient.kt:58]: Removed other catch blocks as any exception will be caught and logged in the top level handler.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/integration/HmppsAuthIntTest.kt [hunk 2]
```diff
@@ -34,17 +36,33 @@ class HmppsAuthIntTest {
   fun `get token service unavailable`() {
     hmppsAuth.stubServiceUnavailableForGetOAuthToken()
 
-    val thrown = assertThrows<RuntimeException> { hmppsAuthGateway.getClientToken() }
+    val ex = assertThrows<SubjectAccessRequestException> { hmppsAuthGateway.getClientToken() }
```
Comment (simon-mitchell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/integration/HmppsAuthIntTest.kt:39]: should this be a full name?

**Review by simon-mitchell — APPROVED** [review]
LGTM

File: src/test/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/integration/HmppsAuthIntTest.kt (hunk 2 above)
Comment (DaveLlewellynMoJ) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/integration/HmppsAuthIntTest.kt:39]: updated

**Review by simon-mitchell — APPROVED** [review]
LGTM

**Pull request merged**